networks/suzuka/setup/src/main.rs

```diff
@@ -15,6 +19,27 @@ async fn main() -> Result<(), anyhow::Error> {
 		)
 		.init();
 
+	let (stop_tx, mut stop_rx) = watch::channel(());
+	tokio::spawn({
+		let mut sigterm = signal(SignalKind::terminate()).context("Can't register to SIGTERM.")?;
+		let mut sigint = signal(SignalKind::interrupt()).context("Can't register to SIGKILL.")?;
+		let mut sigquit = signal(SignalKind::quit()).context("Can't register to SIGKILL.")?;
+		async move {
+			loop {
+				tokio::select! {
+					_ = sigterm.recv() => (),
+					_ = sigint.recv() => (),
+					_ = sigquit.recv() => (),
+				};
+				tracing::info!("Receive Terminate Signal");
+				if let Err(err) = stop_tx.send(()) {
+					tracing::warn!("Can't update stop watch channel because :{err}");
+					return Err::<(), anyhow::Error>(anyhow::anyhow!(err));
+				}
+			}
```

Comment:
This does its job and it's OK, just seems a bit overcomplicated for what it needs to do. A future that does this `select!` once should be enough, and you can await it directly rather than use a watch channel, I think.